wordvecspace/annoy.py

```diff
@@ -85,23 +89,196 @@ def get_distances(self, row_words_or_indices: Union[int, str, np.ndarray],\
+        """Retrives nearest indices when combination=False"""
+
+        if not isinstance(v_w_i, (tuple, list, np.ndarray)):
+            index = self._check_index_or_word(v_w_i)
+
+            nearest_indices, distances = self.ann.get_nns_by_item(index, k, include_distances=True)
+            if include_distances:
+                return (np.array(nearest_indices, dtype=np.uint32), np.array(distances, dtype=np.float32))
+            else:
+                return np.array(nearest_indices, dtype=np.uint32)
+        else:
+            res = list()
+            for item in v_w_i:
+                if isinstance(item, (int, str)):
+                    index = self._check_index_or_word(item)
+                    res.append(self.ann.get_nns_by_item(index, k))
+                else:
+                    res.append(self.ann.get_nns_by_vector(item, k))
+
+            return np.asarray(res, dtype=np.uint32)
+
+    def _get_set_intersect(self, v_w_i, k, examine_k, metric, weights, include_distances):
+        """Performs set_intersect combination method.
+
+        Algorithm:
+        i) retrive nearest indices for each of the item
+        ii) do a set intersection operation
+        iii) find the corresponding distance of each item in interection set
+        iv) retrive nearest indices from intersection set
+        v) return
+        """
+
+        v = self._check_vec(v_w_i)
+        indices = list()
+        distances = list()
+        if not weights:
+            weights = np.ones(len(v))
+        else:
+            weights = np.array(weights)
+        for item in v:
+            idx, dist = self.ann.get_nns_by_vector(item, examine_k, include_distances=True)
+            indices.append(idx)
+            distances.append(dist)
+
+        intersect = np.array(list(set(indices[0]).intersection(*indices)))
+        corresponding_dists = list()
+
+        indices = np.array(indices, dtype=np.uint32)
+        distances = np.array(distances, dtype=np.float32)
+
+        for index in intersect:
+            corresponding_dists.append(self._get_corresponding_dist(v, index, indices, distances, weights))
+
+        corresponding_dists = np.array(corresponding_dists, np.float32)
+        intersect_indices, distances = self._nearest_sorting(corresponding_dists.reshape(1, len(corresponding_dists)), k)
+        nearest_indices = intersect[intersect_indices]
+
+        return (nearest_indices, distances) if include_distances else nearest_indices
+
+    def _get_set_union(self, v_w_i, k, examine_k, metric, weights, include_distances):
+        """Performs set_union combination method.
+
+        Algorithm:
+        i) retrive nearest indices for each item
+        ii) do a set union operation
+        iii) find the corresponding distance of each of the in union set
+        iv) retrieve nearest indices from union set
+        v) return
+        """
+
+        v = self._check_vec(v_w_i)
+        indices = list()
+        distances = list()
+        if not weights:
```

Comment:
You could save a lot of lines and indents if you do these conditions in one line itself:
```
if not weights: weights = np.ones(len(v))
else: weights = np.array(weights)
```

Reply:
noted